### PR TITLE
Support slash commands with enhanced-discord.py

### LINKED ISSUE #83: Support Slash Commands for the new Discord API
In light of [discord.py discontinuation](https://gist.github.com/Rapptz/4a2f62751b9600a31a0d3c78100287f1), it will be necessary to rewrite the bot in a new language or library that supports slash commands.

As mentioned in https://github.com/eibex/reaction-light/issues/59#issuecomment-967019311 there are a few discord.py forks that support slash commands. If none seem suitable and stable the rewrite target language will be Rust with the [Serenity](https://github.com/serenity-rs/serenity) library.

I am not a Rust developer, it will therefore take time for me to learn the language and migrate the code.

### PR DESCRIPTION
**Pull request by eibex**
Supports slash commands by replacing discord.py with [enhanced-discord.py](https://github.com/iDevision/enhanced-discord.py).
This PR is not final and I might still change library or language.

Breaking changes:
- New library
- Python 3.8+ only

Closes #83 

Mutually exclusive PRs:
- #88

### CODE REVIEW

**Comment by eibex** [comment]
@Edwinexd whenever you have time 🙇

File: requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-discord.py>=1.6.0
+disnake>=2.2.1
```
Comment (Edwinexd) [inline requirements.txt:1]:
```suggestion
git+https://github.com/iDevision/enhanced-discord.py@7aafbc2dcf82ec4b8e42b529099c984ff55ffe29
```

File: README.md [hunk 2]
```diff
@@ -36,20 +37,19 @@ You can host the bot yourself by configuring the `config.ini` file (manually or
...
 
 ## Requirements
-This bot requires discord.py and Python 3.6+.
+This bot requires [enhanced-discord.py](https://github.com/iDevision/enhanced-discord.py) and Python 3.8+.
 
-You can get discord.py (and other dependencies) via PyPI:
+You can get enhanced-discord.py via PyPI or GitHub:
 ```
-python3 -m pip install -U discord.py
+python3 -m pip install -U git+https://github.com/iDevision/enhanced-discord.py
```
Comment (Edwinexd) [inline README.md:44]:
```suggestion
python3 -m pip install -r requirements.txt
```

File: README.md [hunk 3]
```diff
@@ -36,20 +37,19 @@ You can host the bot yourself by configuring the `config.ini` file (manually or
 - [License](#license)
 
 ## Requirements
-This bot requires discord.py and Python 3.6+.
+This bot requires [enhanced-discord.py](https://github.com/iDevision/enhanced-discord.py) and Python 3.8+.
 
-You can get discord.py (and other dependencies) via PyPI:
+You can get enhanced-discord.py via PyPI or GitHub:
```
Comment (Edwinexd) [inline README.md:42]:
```suggestion
You can get the required dependencies using PyPI:
```

File: README.md [hunk 4]
```diff
@@ -1,21 +1,22 @@
 # Reaction Light - Discord Role Bot
-[![Reaction Light 2.5.0](https://img.shields.io/badge/Reaction%20Light-2.5.0-yellow.svg)](https://github.com/eibex/reaction-light/blob/master/CHANGELOG.md)
-[![Python 3.6+](https://img.shields.io/badge/python-3.6+-blue.svg)](#)
-[![discord.py 1.6.0+](https://img.shields.io/badge/discord.py-1.6.0+-blue.svg)](#)
+[![Reaction Light 3.0.0](https://img.shields.io/badge/Reaction%20Light-3.0.0-yellow.svg)](https://github.com/eibex/reaction-light/blob/master/CHANGELOG.md)
+[![Python 3.8+](https://img.shields.io/badge/python-3.8+-blue.svg)](#)
+[![enhanced-d.py 1.7.3+](https://img.shields.io/badge/enhanced--d.py-1.7.3+-blue.svg)](#)
```
Comment (Edwinexd) [inline README.md:4]:
```suggestion
[![enhanced-d.py 2.0.0a+](https://img.shields.io/badge/enhanced--d.py-2.0.0a+-blue.svg)](#)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,16 @@
...
+### 3.0.0
+- Support slash commands and drop prefixes ([#83](https://github.com/eibex/reaction-light/issues/83) closed by [#87](https://github.com/eibex/reaction-light/pull/87) by [eibex](https://github.com/eibex))
+- This version is breaking. Version 2.5.0 will continue working until Discord supports the old API (April 2022)
+- README:
+  - What you need to do to make the bot work BEFORE updating:
+    - `python3 -m pip uninstall discord.py`
+    - `python3 -m pip install -U git+https://github.com/iDevision/enhanced-discord.py`
+  - What you need to do to make the bot work AFTER updating:
+    - Kick the bot from your instance
+    - Reinvite it with the following link: `https://discord.com/oauth2/authorize?&client_id=CLIENT_ID&scope=bot%20applications.commands&permissions=2415929408` (replace CLIENT_ID)
```
Comment (Edwinexd) [inline CHANGELOG.md:11]:
```suggestion
    - Reauthorize it with the following link: `https://discord.com/oauth2/authorize?&client_id=CLIENT_ID&scope=bot%20applications.commands&permissions=2415929408` (replace CLIENT_ID)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,16 @@
 # Reaction Light - Changelog
+### 3.0.0
+- Support slash commands and drop prefixes ([#83](https://github.com/eibex/reaction-light/issues/83) closed by [#87](https://github.com/eibex/reaction-light/pull/87) by [eibex](https://github.com/eibex))
+- This version is breaking. Version 2.5.0 will continue working until Discord supports the old API (April 2022)
+- README:
+  - What you need to do to make the bot work BEFORE updating:
+    - `python3 -m pip uninstall discord.py`
+    - `python3 -m pip install -U git+https://github.com/iDevision/enhanced-discord.py`
+  - What you need to do to make the bot work AFTER updating:
+    - Kick the bot from your instance
```
Comment (Edwinexd) [inline CHANGELOG.md:10]:
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,16 @@
 # Reaction Light - Changelog
+### 3.0.0
+- Support slash commands and drop prefixes ([#83](https://github.com/eibex/reaction-light/issues/83) closed by [#87](https://github.com/eibex/reaction-light/pull/87) by [eibex](https://github.com/eibex))
+- This version is breaking. Version 2.5.0 will continue working until Discord supports the old API (April 2022)
+- README:
+  - What you need to do to make the bot work BEFORE updating:
+    - `python3 -m pip uninstall discord.py`
+    - `python3 -m pip install -U git+https://github.com/iDevision/enhanced-discord.py`
```
Comment (Edwinexd) [inline CHANGELOG.md:8]:
```suggestion
    - `python3 -m pip install -r requirements.txt`
```

**Review by Edwinexd — CHANGES_REQUESTED** [review]

**Comment by eibex** [comment]
closing this for now, but not deleting the branch

**Pull request closed without merging**